sdk/program/src/bignum.rs

```diff
@@ -0,0 +1,784 @@
+            {
+                use openssl::bn::BigNum;
+                let value = BigNum::from_u32(val).unwrap().to_vec();
+                Self {
+                    negative: false,
+                    value,
+                }
+            }
+            #[cfg(target_arch = "bpf")]
+            {
+                extern "C" {
+                    fn sol_bignum_from_u32(bytes_ptr: *mut u8, bytes_len: u64, val: u64) -> u64;
+                }
+                let vec_size = {
+                    match val {
+                        0..=255 => 1,
+                        256..=65_535 => 2,
+                        65_536..=16_777_215 => 3,
+                        _ => 4,
+                    }
+                };
+                let mut value = Vec::<u8>::with_capacity(vec_size);
+                unsafe {
+                    sol_bignum_from_u32(
+                        value.as_mut_ptr() as *mut _ as *mut u8,
+                        vec_size as u64,
+                        val as u64,
+                    );
+                    value.set_len(vec_size);
+                }
+                Self {
+                    negative: false,
+                    value,
+                }
+            }
+        }
+    }
+    /// Returns a BigNumber from a decimal string
+    pub fn from_dec_str(string: &str) -> Self {
+        #[cfg(not(target_arch = "bpf"))]
+        {
+            use openssl::bn::BigNum;
+            let bn = BigNum::from_dec_str(string).unwrap();
+            let value = bn.to_vec();
+            Self {
+                negative: bn.is_negative(),
+                value,
+            }
+        }
+        #[cfg(target_arch = "bpf")]
+        {
+            extern "C" {
+                fn sol_bignum_from_dec_str(
+                    in_dec_str_ptr: *const u64,
+                    in_size: u64,
+                    out_ptr: *mut u8,
+                    out_size_ptr: *mut u64,
+                    out_is_neg_flag_ptr: *mut u64,
+                ) -> u64;
+            }
+            let mut value = Vec::<u8>::with_capacity(string.len());
+            let mut value_len = string.len() as u64;
+            let mut is_negative = 0u64;
+            unsafe {
+                sol_bignum_from_dec_str(
+                    string.as_ptr() as *const _ as *const u64,
+                    string.len() as u64,
+                    value.as_mut_ptr() as *mut _ as *mut u8,
+                    &mut value_len as *mut _ as *mut u64,
+                    &mut is_negative as *mut _ as *mut u64,
+                );
+                value.set_len(value_len as usize);
+            }
+            Self {
+                negative: is_negative != 0,
+                value,
+            }
+        }
+    }
+
+    /// Flag indicating if negative or not
+    pub fn is_negative(&self) -> bool {
+        self.negative
+    }
+
+    /// Returns an array of bytes of self
+    pub fn to_bytes(&self) -> &[u8] {
```

Comment:
`as_bytes`

Reply:
Fixed in a156563ebf54e8934b207cb12cb2d8aefb83b93d